enterprise_gateway/services/processproxies/k8s.py

```diff
@@ -310,14 +326,66 @@ def _create_kernel_namespace(self, service_account_name: str) -> str:
                     body = client.V1DeleteOptions(
                         grace_period_seconds=0, propagation_policy="Background"
                     )
-                    client.CoreV1Api().delete_namespace(name=namespace, body=body)
+                    client.CoreV1Api(api_client=kubernetes_client).delete_namespace(
+                        name=namespace, body=body
+                    )
                     self.log.warning(f"Deleted kernel namespace: {namespace}")
                 else:
                     reason = f"Error occurred creating namespace '{namespace}': {err}"
                 self.log_and_raise(http_status_code=500, reason=reason)
 
         return namespace
 
+    def _create_service_account_if_not_exists(
+        self, namespace: str, service_account_name: str
+    ) -> None:
+        """If service account doesn't exist in target cluster, create one. Occurs if a remote cluster is being used."""
+        service_account_list_in_namespace: client.V1ServiceAccountList = client.CoreV1Api(
+            api_client=kubernetes_client
+        ).list_namespaced_service_account(namespace=namespace)
+
+        service_accounts_in_namespace: list[
+            client.V1ServiceAccount
+        ] = service_account_list_in_namespace.items
+        service_account_names_in_namespace: list[str] = [
+            svcaccount.metadata.name for svcaccount in service_accounts_in_namespace
+        ]
+
+        if service_account_name not in service_account_names_in_namespace:
+            service_account_metadata = {"name": service_account_name}
+            service_account_to_create: client.V1ServiceAccount = client.V1ServiceAccount(
+                kind="ServiceAccount", metadata=service_account_metadata
+            )
+
+            client.CoreV1Api(api_client=kubernetes_client).create_namespaced_service_account(
+                namespace=namespace, body=service_account_to_create
+            )
+
+            self.log.info(
+                f"Created service account {service_account_name} in namespace {namespace}"
```

Comment:
Is there any way to access the "name" of the external cluster?  Seems really helpful to include that here.
```suggestion
                f"Created service account {service_account_name} in namespace {namespace} of external cluster {external_cluster}"
```